code/datums/status_effects/buffs.dm

```diff
@@ -228,3 +228,30 @@
 	name = "Wish Granter's Immortality"
 	desc = "You are being resurrected!"
 	icon_state = "wish_granter"
+
+/datum/status_effect/cult_master
+	id = "The Cult Master"
+	duration = -1
+	tick_interval = 100
+	alert_type = null
+	var/alive = TRUE
+
+/datum/status_effect/cult_master/tick()
+	if(!owner)
+		src.Destroy()
+		return
+	if(owner.stat != DEAD && !alive)
+		alive = TRUE
+		return
+	if(owner.stat == DEAD && alive)
+		alive = FALSE
+		on_remove()
```

Comment:
and when you've done that, call `remove_message_proc()` here instead of on_remove(), because uses of on_remove assume that it'll call . = ..(), which will call qdel(src)